packages/cron/src/jobs/goodbits.js

```diff
@@ -0,0 +1,137 @@
+
+export async function sync({ env }) {
+  const cfApiToken = mustGetEnv('CF_API_TOKEN')
+  const ghToken = mustGetEnv('GH_TOKEN')
+
+  const wranglerConfig = await getWranglerToml(wranglerConfigPath)
+  const wranglerEnvConfig = wranglerConfig.env[env]
+  if (!wranglerEnvConfig) {
+    throw new Error(`missing wrangler configuration for env: ${env}`)
+  }
+  console.log(`🧩 using wrangler config: ${wranglerConfigPath}`)
+
+  const cfAccountId = wranglerEnvConfig.account_id
+  if (!cfAccountId) {
+    throw new Error(`missing Cloudflare account_id in env: ${env}`)
+  }
+  console.log(`🏕 using env: ${env} (${cfAccountId})`)
+
+  const kvNamespaces = wranglerEnvConfig.kv_namespaces || []
+  const goodbitsListKv = kvNamespaces.find(
+    (kv) => kv.binding === 'GOODBITSLIST'
+  )
+  if (!goodbitsListKv) {
+    throw new Error('missing binding in kv_namespaces: GOODBITSLIST')
+  }
+  console.log(`🪢 using KV binding: GOODBITSLIST (${goodbitsListKv.id})`)
+
+  for (const url of GOODBITS_SOURCES) {
+    console.log(`🦴 fetching ${url}`)
+    const goodbitsList = await getGoodbitsList(url, ghToken)
+    console.log('goodbits', goodbitsList)
+    const kvs = goodbitsList.map(({ cid, tags }) => ({
+      key: cid,
+      value: { tags },
+    }))
+
+    console.log(`📝 writing ${kvs.length} entries`)
+    await writeKVMulti(cfApiToken, cfAccountId, goodbitsListKv.id, kvs)
+  }
+
+  console.log('✅ Done')
+}
+
+/**
+ * @param {string} apiToken Cloudflare API token
+ * @param {string} accountId Cloudflare account ID
+ * @param {string} nsId KV namespace ID
+ * @param {Array<BulkWritePair>} kvs
+ * @returns {Promise<void>}
+ */
+async function writeKVMulti(apiToken, accountId, nsId, kvs) {
+  const url = `https://api.cloudflare.com/client/v4/accounts/${accountId}/storage/kv/namespaces/${nsId}/bulk`
+  kvs = kvs.map((kv) => ({
+    ...kv,
+    value: JSON.stringify(kv.value),
+  }))
+
+  const chunkSize = 10000
+  for (let i = 0; i < kvs.length; i += chunkSize) {
+    const kvsChunk = kvs.slice(i, i + chunkSize)
+    const res = await fetch(url, {
+      method: 'PUT',
+      headers: {
+        Authorization: `Bearer ${apiToken}`,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(kvsChunk),
+    })
+    const { success, errors } = await res.json()
+    if (!success) {
+      const error = Array.isArray(errors) && errors[0]
+      throw new Error(
+        error ? `${error.code}: ${error.message}` : 'failed to write to KV'
+      )
+    }
+  }
+}
+
+/**
+ * @param {string} url
+ * @param {string} ghToken
+ */
+async function getGoodbitsList(url, ghToken) {
+  const headers = new Headers()
+  headers.append('authorization', `token ${ghToken}`)
+  headers.append('cache-control', 'no-cache')
+  headers.append('pragma', 'no-cache')
+
+  const res = await fetch(url, {
+    headers,
+  })
+  if (!res.ok) {
+    throw new Error(`unexpected status fetching goodbits list: ${res.status}`)
+  }
+  const list = ndjsonParser(await res.text())
```

Comment:
_y u no support your friendly neighbourhood software developers‽_
- https://www.npmjs.com/package/iterable-ndjson for modern iterable flavour goodness from alan
- https://www.npmjs.com/package/ndjson the original streaming ndjson from contra & max

Reply:
I didn't know about these packages :) Chose the one with better score in `npms`, but I will change to Alan's one